### DISCUSSION TITLE
Create a check for a `workflow_run` triggered workflow

### DISCUSSION THREAD

**Discussion by blt** [community/community#38379] [Actions] [answered]
I have a workflow that must be split into an untrusted -- that is, no access to repo secrets -- and a trusted component with a `workflow_run` trigger. I need to achieve two things in this split:

* the trusted workflow must be able to post a comment on PRs and
* we need to block merging of PRs depending on the success/failure of the trusted workflow.

Achieving the first was straightforward following [available documentation](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#using-data-from-the-triggering-workflow). From that link, with the addition of trust details:

```
name: Upload data (untrusted)

on:
  pull_request:

jobs:
  upload:
    runs-on: ubuntu-latest

    steps:
      - name: Save PR number
        env:
          PR_NUMBER: ${{ github.event.number }}
        run: |
          mkdir -p ./pr
          echo $PR_NUMBER > ./pr/pr_number
      - uses: actions/upload-artifact@v3
        with:
          name: pr_number
          path: pr/
```

```
name: Use the data (trusted)

on:
  workflow_run:
    workflows: ["Upload data (untrusted)"]
    types:
      - completed

jobs:
  download:
    runs-on: ubuntu-latest
    steps:
      - name: 'Download artifact'
        uses: actions/github-script@v6
        with:
          script: |
            let allArtifacts = await github.rest.actions.listWorkflowRunArtifacts({
               owner: context.repo.owner,
               repo: context.repo.repo,
               run_id: context.payload.workflow_run.id,
            });
            let matchArtifact = allArtifacts.data.artifacts.filter((artifact) => {
              return artifact.name == "pr_number"
            })[0];
            let download = await github.rest.actions.downloadArtifact({
               owner: context.repo.owner,
               repo: context.repo.repo,
               artifact_id: matchArtifact.id,
               archive_format: 'zip',
            });
            let fs = require('fs');
            fs.writeFileSync(`${process.env.GITHUB_WORKSPACE}/pr_number.zip`, Buffer.from(download.data));

      - name: 'Unzip artifact'
        run: unzip pr_number.zip

      - name: 'Comment on PR'
        uses: actions/github-script@v6
        with:
          github-token: ${{ secrets.GITHUB_TOKEN }}
          script: |
            let fs = require('fs');
            let issue_number = Number(fs.readFileSync('./pr_number'));
            await github.rest.issues.createComment({
              owner: context.repo.owner,
              repo: context.repo.repo,
              issue_number: issue_number,
              body: 'Thank you for the PR!'
            });
```

The second goal of posting a check and updating it as the untrusted workflow moves along has proved more difficult. I tried something like this in the trusted workflow:

```
  mark-check-in-progress:
    name: Mark check in-progress on originating PR
    runs-on: ubuntu-22.04
    needs:
      - compute-metadata
    steps:
      - name: 'Set check status in-progress'
        uses: actions/github-script@v3.1.0
        with:
          script: |
            github.rest.checks.create({
               owner: context.repo.owner,
               repo: context.repo.repo,
               name: "Use the data (trusted)",
               head_sha: github.event.workflow_run.head_sha,
               status: "in_progress",
               output: {
                  title: "Use the data (trusted)",
                  summary: "Using data from untrusted workflow, won't be a minute.",
                  text: '',
               },
            });
```

I end up being greeted with this error:

```
> Run actions/github-script@v3.1.0
TypeError: Cannot read property 'checks' of undefined
Error: Unhandled error: TypeError: Cannot read property 'checks' of undefined
    at eval (eval at callAsyncFunction (/home/runner/work/_actions/actions/github-script/v3.1.0/dist/index.js:5368:56), <anonymous>:1:13)
    at callAsyncFunction (/home/runner/work/_actions/actions/github-script/v3.1.0/dist/index.js:5369:12)
    at main (/home/runner/work/_actions/actions/github-script/v3.1.0/dist/index.js:5394:26)
    at Module.720 (/home/runner/work/_actions/actions/github-script/v3.1.0/dist/index.js:5378:1)
    at __webpack_require__ (/home/runner/work/_actions/actions/github-script/v3.1.0/dist/index.js:24:31)
    at startup (/home/runner/work/_actions/actions/github-script/v3.1.0/dist/index.js:43:19)
    at /home/runner/work/_actions/actions/github-script/v3.1.0/dist/index.js:49:18
    at Object.<anonymous> (/home/runner/work/_actions/actions/github-script/v3.1.0/dist/index.js:52:10)
    at Module._compile (internal/modules/cjs/loader.js:999:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
```

I'm not any good at javascript -- I'm trying to work solely from the Github documentation here -- so it's not clear to me why 'checks' would be undefined or what that means particularly, or even if the approach I've attempted in `mark-check-in-progress` is vaguely in the right direction or even if it's possible for a check to be made from a `workflow_run` triggered workflow at all. 

Any advice would be greatly appreciated.

**Comment by ossanna16** [comment]
Hi there @blt and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our official introduction thread
To connect with other new users, check out this Discussion to Meet Your GitHub Community Team, further your GitHub knowledge by working through our GitHub Skill 🆙 exercises, and participate in our weekly Fun Friday chats

**Comment by blt** [comment, marked as answer]
The problem here, ultimately, is that some of Github's documentation references an older version of github-script, the version given in the above question. As of this writing v6.3 is current and the API has shifted non-trivially.

**Comment by metecanpolat1** [comment]
@blt

**Reply by metecanpolat1** [reply]
@metecanpolat1

**Comment by kdubb** [comment]
@blt Did you get this working?

In my case the check creation succeeds but the check doesn't show up on the original PR (we targeting `github.event.workflow_run.head_sha` as well) or anywhere else.

There is a weird version of the check attached to the workflow triggered by `workflow_run` but it's even missing details (like the title).

**Comment by blt** [comment]
Hi Kevin, I did not. Github never contacted me either. I believe this is just broken, at least when I looked at it last. 

- blt

**Comment by kdubb** [comment]
Can you revoke your "answer"? Would help in trying to get eyes on this.

**Comment by blt** [comment]
It is an answer to my satisfaction. I don’t have any interest in pursuing this further. 

- blt
